db-service/lib/cqn2sql.js

```diff
@@ -324,7 +324,7 @@ class CQN2SQLRenderer {
       DistanceFromRoot: { xpr: [{ ref: ['HIERARCHY_LEVEL'] }, '-', { val: 1, param: false }], as: 'DistanceFromRoot' },
       DrillState: false,
       LimitedDescendantCount: { xpr: [{ ref: ['HIERARCHY_TREE_SIZE'] }, '-', { val: 1, param: false }], as: 'LimitedDescendantCount' },
-      LimitedRank: { xpr: [{ func: 'row_number', args: [], xpr: ['OVER', { list: [] }] }, '-', { val: 1, param: false }], as: 'LimitedRank' }
+      LimitedRank: { xpr: [{ func: 'row_number', args: [] }, 'OVER', { list: [] }, '-', { val: 1, param: false }], as: 'LimitedRank' }
```

Comment:
```suggestion
      LimitedRank: { xpr: [{ func: 'row_number', args: [] }, 'OVER', { xpr: [] }, '-', { val: 1, param: false }], as: 'LimitedRank' }
```

Lets use `xpr` as an empty list is invalid for `SQL`. We should probably keep the fallbacks in the HANAService. Also the `{func,xpr}` syntax should work for HANA as well. If it doesn't that should be fixed as well.